assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,328 @@
+	let store;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createRegistry();
+
+		storeDefinition = createFetchStore( STORE_PARAMS );
+		registry.registerStore( STORE_NAME, storeDefinition );
+		dispatch = registry.dispatch( STORE_NAME );
+		store = registry.stores[ STORE_NAME ].store;
+		select = registry.select( STORE_NAME );
+
+		apiFetchSpy = jest.spyOn( { apiFetch }, 'apiFetch' );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'actions', () => {
+		it( 'includes the expected actions', () => {
+			const fetchStoreDefinition = createFetchStore( {
+				baseName: 'SaveSomeData',
+				controlCallback: async () => true,
+				reducerCallback: ( state ) => state,
+			} );
+
+			expect( Object.keys( fetchStoreDefinition.actions ) ).toEqual( [
+				'fetchSaveSomeData',
+				'receiveSaveSomeData',
+			] );
+		} );
+
+		describe( 'fetch', () => {
+			it( 'validates parameters based on argsToParams', () => {
+				const consoleErrorSpy = jest.spyOn( global.console, 'error' );
+
+				dispatch.fetchGetSomeData();
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'objParam is required.' );
+
+				dispatch.fetchGetSomeData( 123 );
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'objParam is required.' );
+
+				dispatch.fetchGetSomeData( {} );
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'aParam is required.' );
+
+				consoleErrorSpy.mockClear();
+			} );
+
+			it( 'yields the expected actions for an arguments error', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+					argsToParams: ( requiredParam ) => {
+						invariant( requiredParam, 'requiredParam is required.' );
+						return {
+							requiredParam,
+						};
+					},
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				// Catch invariant to get exactly the error we expect.
+				let error;
+				try {
+					invariant( false, 'requiredParam is required.' );
+				} catch ( err ) {
+					error = err;
+				}
+
+				expect( action.next().value ).toEqual( {
+					response: undefined,
+					error,
+				} );
+			} );
+
+			it( 'yields the expected actions for a success request', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				expect( action.next().value.type ).toEqual( 'START_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FETCH_SAVE_SOME_DATA' );
+				expect( action.next( 42 ).value.type ).toEqual( 'RECEIVE_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FINISH_FETCH_SAVE_SOME_DATA' );
```

Comment:
For these I think we could list out the actions to ensure the order is correct while keeping the assertions separate.
E.g.
```js
const [ startAction, fetchAction, receiveAction, finishAction ] = action;
expect( startAction ) // ...;
expect( fetchAction ) // ...;
//...
```

Reply:
See above.